packages/network-controller/src/NetworkController.ts

```diff
@@ -1939,6 +1975,11 @@ export class NetworkController extends BaseController<
       buildNetworkConfigurationsByNetworkClientId(
         this.state.networkConfigurationsByChainId,
       );
+
+    this.messagingSystem.publish(
+      'NetworkController:networkRemoved',
```

Comment:
This seems redundant - the state change event already tells us when networks are removed, doesn't it?

Reply:
Hmm, I can see a possible reason why this was added. There's already a `networkAdded` event. This was ported over from the previous version of `addNetwork` to allow consumers to create MetaMetrics events when a network was added. So perhaps this event is being added for symmetry? Perhaps it should not have been added, however.

Reply:
@Gudahtt @mcmire 
My only concern is if `stateChanged` event is too spammy. And would this state change also include the "previous state" of the network that was removed?

If so, then I can investigate removing this event (we added it since there was an existing `networkAdded` event.)

Reply:
I've investigated the patch calls, this does not contain a `value` property. So I don't see a way of using the `stateChange` event to get the underlying deleted `NetworkConfiguration`.

Reply:
The patch is not meant to be used in the way it is used in the snippet that you've shared, I'd strongly discourage that. There are multiple ways for state to change, and relying on specific patch operations can result in bugs (e.g. checking for `remove` operations but not `replace`, or checking at the wrong level).

Reply:
The `selector` function is how you can narrow down state changes to just those that you care about: https://github.com/MetaMask/core/blob/54bc0f4348b55e6f148d09ebbd9e4a646ea098c6/packages/base-controller/src/ControllerMessenger.ts#L316

Reply:
@Prithpal-Sooriya Correct me if I'm wrong, but I suspect you may need access to the network that got removed so that you can apply that removal to other clients, is that the motivation for adding this event?

Reply:
@mcmire yep exactly, we need the network that was removed, not just the chainId.

If we were to use the existing `stateChanged` event, I don't see where we can get this removed network.

Reply:
Ah OK. We can handle that with just the state change event + selector as well, but we'd need to cache a list of existing network IDs on the subscriber side which is definitely not ideal.

Adding the remove event makes sense